grammar/TypeQL.g4

```diff
@@ -102,7 +102,11 @@ pattern_negation      :   NOT '{' patterns '}'                        ;
 
 pattern_variable      :   variable_concept
                       |   variable_type
-                      |   variable_thing_any   ;
+                      |   variable_thing_any
+                      |   variable_value
+                      ;
+
+either_var            :   VAR_            |   VALVAR_         ;
```

Comment:
VAR_ should be VAR_NAMED 
and either_var should go with the variables to as a non-terminal named NAME_OR_VALUE